toolchain/check/context.cpp

```diff
@@ -1221,6 +1230,17 @@ auto Context::GetTypeIdForTypeConstant(SemIR::ConstantId constant_id)
   return SemIR::TypeId::ForTypeConstant(constant_id);
 }
 
+auto Context::FacetTypeFromInterface(SemIR::InterfaceId interface_id,
+                                     SemIR::SpecificId specific_id)
+    -> SemIR::FacetType {
+  llvm::SmallVector<SemIR::FacetTypeInfo::Impls> impls;
+  impls.emplace_back(interface_id, specific_id);
+  SemIR::FacetTypeId facet_type_id =
+      sem_ir().facet_types().Add(SemIR::FacetTypeInfo{
+          .impls = impls, .requirement_block_id = SemIR::InstBlockId::Invalid});
```

Comment:
Would something like this work instead? If so, I think I might find it a little clearer because it's more declarative.

```suggestion
  SemIR::FacetTypeId facet_type_id =
      sem_ir().facet_types().Add(SemIR::FacetTypeInfo{
          .impls = {{interface_id, specific_id}}, .requirement_block_id = SemIR::InstBlockId::Invalid});
```